code/obj/item/material.dm

```diff
@@ -882,19 +882,41 @@
 
 		playsound(src.loc, sound_process, 40, 1)
 
+	proc/load_reclaim(obj/item/W as obj, mob/user as mob)
+		. = FALSE
+		if (istype(W,/obj/item/raw_material/) || istype(W,/obj/item/sheet/) || istype(W,/obj/item/rods/) || istype(W,/obj/item/tile/) || istype(W,/obj/item/cable_coil) || istype(W,/obj/item/wizard_crystal))
+			W.set_loc(src)
+			if (user) user.u_equip(W)
+			W.dropped()
+			. = TRUE
+
 	attackby(obj/item/W as obj, mob/user as mob)
 		if (W.cant_drop) //For borg held items
 			boutput(user, "<span class='alert'>You can't put that in [src] when it's attached to you!</span>")
 			return ..()
-		if (istype(W,/obj/item/raw_material/) || istype(W,/obj/item/sheet/) || istype(W,/obj/item/rods/) || istype(W,/obj/item/tile/) || istype(W,/obj/item/cable_coil) || istype(W,/obj/item/wizard_crystal))
+
+		if (istype(W,/obj/item/storage/) || istype(W,/obj/item/satchel/))
+			var/obj/item/storage/S = W
+			var/obj/item/satchel/B = W
+			for(var/obj/item/O in W)
+				if (load_reclaim(O))
+					. = TRUE
+					if (istype(S))
+						S.hud.remove_object(O)
+					else if (istype(B))
+						B.satchel_updateicon()
```

Comment:
why call this constantly, and not just at the end, this could get called a lot very fast

Reply:
Good point